lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+            order_string,
+            limit_string
+          ].delete_if(&:blank?) * ' '
+        end
+
+        def select_string
+          if @scope.select_values.any?
+            "id, jsonb_slice(attribute_store, #{@adapter.fields_to_postgres_array(@scope.select_values)}) as attribute_store"
+          else
+            '*'
+          end
+        end
+
+        def where_string
+          wheres = @scope.where_values.dup
+          if @scope.id_values.any?
+            wheres << @adapter.create_ids_where_clause(@scope.id_values)
+          end
+
+          if wheres.any?
+            "WHERE #{wheres * ' AND '}"
+          end
+        end
+
+        def order_string
+          if @scope.order_values.any?
+            orders = @scope.order_values.join(', ')
+            "ORDER BY #{orders}"
+          elsif @scope.id_values.many?
+            id_orders = @scope.id_values.map { |id| "ID=#{@adapter.quote(id)} DESC" }.join(',')
+            "ORDER BY #{id_orders}"
+          end
+        end
+
+        def limit_string
+          if @scope.limit_value
+            "LIMIT #{@scope.limit_value}"
+          end
+        end
+      end
+
+      def primary_key_column
+        'id'
+      end
+
+      def connection
+        # conf = {:adapter=>"postgresql", :encoding=>"unicode", :database=>"axle_place_test", :pool=>5, :username=>"postgres"}
+        # @connection ||= ActiveRecord::Base.postgresql_connection(conf)
+        ActiveRecord::Base.connection
+      end
+
+      def execute(statement)
+        ActiveSupport::Notifications.instrument("cql.cassandra_object", cql: statement) do
+          connection.execute statement
+        end
+      end
+
+      def select(scope)
+        statement = QueryBuilder.new(self, scope).to_query
+
+        connection.execute(statement).each do |attributes|
+          yield attributes[primary_key_column], Oj.compat_load(attributes['attribute_store'])
+        end
+      end
+
+      def insert(table, id, attributes)
+        not_nil_attributes = attributes.reject { |key, value| value.nil? }
+        statement = "INSERT INTO #{table} (#{primary_key_column}, attribute_store) VALUES (#{quote(id)}, #{to_quoted_jsonb(not_nil_attributes)})"
+        execute_batchable statement
+      end
+
+      def update(table, id, attributes)
+        return if attributes.empty?
+
+        not_nil_attributes = attributes.reject { |key, value| value.nil? }
+        nil_attributes = attributes.select { |key, value| value.nil? }
+
+        if not_nil_attributes.any? && nil_attributes.any?
+          value_update = "jsonb_merge(jsonb_delete(attribute_store, #{fields_to_postgres_array(nil_attributes.keys)}), #{to_quoted_jsonb(not_nil_attributes)})"
+        elsif not_nil_attributes.any?
+          value_update = "jsonb_merge(attribute_store, #{to_quoted_jsonb(not_nil_attributes)})"
+        elsif nil_attributes.any?
+          value_update = "jsonb_delete(attribute_store, #{fields_to_postgres_array(nil_attributes.keys)})"
```

Comment:
It might also be smart to include the name of the schema when calling the functions.  In the create function statements, the functions are explicitly created in the 'public' schema.  I don't know if postgresql always looks in the public schema, but if the data tables were in the 'foo' schema, then you would want to call the function as `public.jsonb_delete`.

Overall, I'm not sure that rails/postgres really has good support for using a schema other than public, so this is probably a non-issue.